packages/astro/test/units/dev/collections-mixed-content-errors.test.js

```diff
@@ -1,129 +1,146 @@
 import * as assert from 'node:assert/strict';
 import { describe, it } from 'node:test';
-import { fileURLToPath } from 'node:url';
 import _sync from '../../../dist/core/sync/index.js';
-import { createFsWithFallback } from '../test-utils.js';
+import { createFixture } from '../test-utils.js';
 
-const root = new URL('../../fixtures/content-mixed-errors/', import.meta.url);
-
-async function sync({ fs }) {
+async function sync(root) {
 	try {
-		await _sync(
-			{
-				root: fileURLToPath(root),
-				logLevel: 'silent',
-			},
-			{
-				fs,
-			},
-		);
+		await _sync({
+			root,
+			logLevel: 'silent',
+		});
 		return 0;
-	} catch (_) {
+	} catch {
 		return 1;
 	}
 }
 
+const baseFileTree = {
+	'/src/content/authors/placeholder.json': `{ "name": "Placeholder" }`,
+	'/src/content/blog/placeholder.md': `\
+---
+title: Placeholder post
+---
+`,
+	'/src/pages/authors.astro': `\
+---
+import { getCollection } from 'astro:content';
+try {
+	await getCollection('authors')
+} catch (e) {
+	return e
+}
+---
+
+<h1>Worked</h1>
+`,
+	'/src/pages/blog.astro': `\
+---
+import { getCollection } from 'astro:content';
+
+await getCollection('blog')
+---
+
+<h1>Worked</h1>`,
+};
+
 describe('Content Collections - mixed content errors', () => {
 	it('raises "mixed content" error when content in data collection', async () => {
-		const fs = createFsWithFallback(
-			{
-				'/src/content/authors/ben.md': `---
+		const fixture = await createFixture({
+			...baseFileTree,
+			'/src/content/authors/ben.md': `\
+---
 name: Ben
 ---
 
-# Ben`,
-				'/src/content/authors/tony.json': `{ "name": "Tony" }`,
-				'/src/content/config.ts': `
-
-					import { z, defineCollection } from 'astro:content';
```

Comment:
I tried to dedent the code so that it looks good in the file written, but there's too many so I gave up halfway through. It helps reduce diff for now too.